internal/service/api/responses/get_promocode.go

```diff
@@ -1,10 +1,14 @@
 package responses
 
 import (
-	"gitlab.com/tokend/nft-books/generator-svc/internal/data"
-	"gitlab.com/tokend/nft-books/generator-svc/resources"
+	"github.com/dl-nft-books/core-svc/internal/data"
+	"github.com/dl-nft-books/core-svc/resources"
 )
 
-func NewGetPromocodeResponse(promocode data.Promocode) *resources.PromocodeResponse {
-	return &resources.PromocodeResponse{Data: promocode.Resource()}
+func NewGetPromocodeResponse(promocode data.Promocode) (*resources.PromocodeResponse, error) {
+	response, err := promocode.Resource()
```

Comment:
Convert it explicitly here and remove `Resource` method from the promo code structure

Reply:
Fixed